inc/class-local-server-extension.php

```diff
@@ -0,0 +1,246 @@
+			];
+		}
+
+		// Enable Kibana. (Defaults to true, but supports new + old style.)
+		$has_kibana = $local_config['kibana'] ?? $full_config['local-server']['kibana'] ?? true;
+		if ( $has_kibana ) {
+			if ( ! empty( $full_config['local-server']['kibana'] ) ) {
+				trigger_error(
+					'extra.altis.modules.local-server.kibana is deprecated, use extra.altis.modules.search.kibana instead.',
+					E_USER_DEPRECATED
+				);
+			}
+
+			$config['services'] = array_merge( $config['services'], $this->get_service_kibana() );
+		}
+
+		return $config;
+	}
+
+	/**
+	 * Get the Elasticsearch service.
+	 *
+	 * @return array
+	 */
+	protected function get_service_elasticsearch() : array {
+		$mem_limit = getenv( 'ES_MEM_LIMIT' ) ?: '1g';
+
+		$version_map = [
+			'7.10' => 'humanmade/altis-local-server-elasticsearch:4.1.0',
+			'7' => 'humanmade/altis-local-server-elasticsearch:4.1.0',
+			'6.8' => 'humanmade/altis-local-server-elasticsearch:3.1.0',
+			'6' => 'humanmade/altis-local-server-elasticsearch:3.1.0',
+			'6.3' => 'humanmade/altis-local-server-elasticsearch:3.0.0',
+		];
+
+		$this->check_elasticsearch_version( array_keys( $version_map ) );
+
+		$image = $version_map[ $this->get_elasticsearch_version() ];
+
+		return [
+			'elasticsearch' => [
+				'image' => $image,
+				'restart' => 'unless-stopped',
+				'container_name' => "{$this->generator->project_name}-es",
+				'ulimits' => [
+					'memlock' => [
+						'soft' => -1,
+						'hard' => -1,
+					],
+				],
+				'mem_limit' => $mem_limit,
+				'volumes' => [
+					'es-data:/usr/share/elasticsearch/data',
+					"{$this->generator->root_dir}/content/uploads/es-packages:/usr/share/elasticsearch/config/packages",
+				],
+				'ports' => [
+					'9200',
+				],
+				'networks' => [
+					'proxy',
+					'default',
+				],
+				'healthcheck' => [
+					'test' => [
+						'CMD-SHELL',
+						'curl --silent --fail localhost:9200/_cluster/health || exit 1',
+					],
+					'interval' => '5s',
+					'timeout' => '5s',
+					'retries' => 25,
+				],
+				'labels' => [
+					'traefik.port=9200',
+					'traefik.protocol=http',
+					'traefik.docker.network=proxy',
+					"traefik.frontend.rule=HostRegexp:elasticsearch-{$this->generator->hostname}",
+					"traefik.domain=elasticsearch-{$this->generator->hostname}",
+				],
+				'environment' => [
+					'http.max_content_length=10mb',
+					// Force ES into single-node mode (otherwise defaults to zen discovery as
+					// network.host is set in the default config).
+					'discovery.type=single-node',
+					// Use max container memory limit as the max JVM heap allocation value.
+					"ES_JAVA_OPTS=-Xms512m -Xmx{$mem_limit}",
+				],
+			],
+		];
+	}
+
+	/**
+	 * Get the Kibana service.
+	 *
+	 * @return array
+	 */
+	protected function get_service_kibana() : array {
+		$version_map = [
+			'7.10' => 'humanmade/altis-local-server-kibana:1.1.1',
+			'7' => 'humanmade/altis-local-server-kibana:1.1.1',
+			'6.8' => 'blacktop/kibana:6.8',
```

Comment:
A thought. Why are we still supporting so many old versions? Is this just for clients who haven't upgraded? Or is there a technical reason someone would want to use an old version?